### PR TITLE
Fix <Admin theme> needs createTheme

### PR DESCRIPTION
**Pull request by fzaninotto**

### CODE REVIEW

File: docs/Theming.md [hunk 1]
```diff
@@ -251,16 +251,14 @@ export const PostList = () => {
 
 ## Using a Predefined Theme
 
-Material UI also supports [complete theming](https://material-ui.com/customization/themes) out of the box. Material UI ships two base themes: light and dark. React-admin uses the light one by default. To use the dark one, pass it to the `<Admin>` component, in the `theme` prop (along with `createTheme()`).
+Material UI also supports [complete theming](https://material-ui.com/customization/themes) out of the box. Material UI ships two base themes: light and dark. React-admin uses the light one by default. To use the dark one, pass it to the `<Admin>` component, in the `theme` prop.
```
Comment (djhi) [inline docs/Theming.md:254]: We probably should mention that using `createTheme` is fine too

File: examples/demo/src/App.tsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ const App = () => {
             )}
             authProvider={authProvider}
             dashboard={Dashboard}
-            loginPage={Login}
+            loginPage={<Login />}
```
Comment (djhi) [inline examples/demo/src/App.tsx:40]: This is weird

Comment (fzaninotto) [reply examples/demo/src/App.tsx]: It does work, though. I removed it so your eyes don't bleed.

File: docs/Theming.md (hunk 1 above)
Comment (fzaninotto) [reply docs/Theming.md:254]: I think it would be confusing for newcomers.

File: packages/ra-core/src/types.ts [hunk 3]
```diff
@@ -278,10 +278,7 @@ export type AdminChildren = RenderResourcesFunction | ReactNode;
 export type TitleComponent = string | ReactElement<any>;
 export type CatchAllComponent = ComponentType<{ title?: TitleComponent }>;
 
-export interface LoginComponentProps {
-    title?: TitleComponent;
-}
-export type LoginComponent = ComponentType<LoginComponentProps>;
+export type LoginComponent = ComponentType<{}> | ReactElement<any>;
 export type DashboardComponent = ComponentType<WithPermissionsChildrenParams>;
```
Comment (djhi) [inline packages/ra-core/src/types.ts:282]: Shouldn't we do the same for the Dashboard ?

Comment (fzaninotto) [reply packages/ra-core/src/types.ts:282]: yes, in due time

**Pull request merged**